pkg/resources/pullSecret.go

```diff
@@ -85,3 +72,43 @@ func ReconcileSecretToRHMIOperatorNamespace(ctx context.Context, client k8sclien
 
 	return integreatlyv1alpha1.PhaseCompleted, nil
 }
+
+func LinkSecretToServiceAccounts(ctx context.Context, client k8sclient.Client, namespace string, secretName string) error {
+	serviceAccounts := &corev1.ServiceAccountList{}
+	listOpts := []k8sclient.ListOption{
+		k8sclient.InNamespace(namespace),
+	}
+	err := client.List(ctx, serviceAccounts, listOpts...)
+	if err != nil {
+		return err
+	}
+
+	for _, sa := range serviceAccounts.Items {
+		currentSa := &corev1.ServiceAccount{}
+		err = client.Get(ctx, types.NamespacedName{Namespace: namespace, Name: sa.Name}, currentSa)
+		if err != nil {
+			return err
+		}
+
+		pullSecretFound := false
+		for _, ips := range currentSa.ImagePullSecrets {
+			if ips.Name == secretName {
+				pullSecretFound = true
+				break
+			}
+
+		}
+
+		if !pullSecretFound {
+			newPullSecret := corev1.LocalObjectReference{Name: secretName}
+			_, err = controllerutil.CreateOrUpdate(ctx, client, currentSa, func() error {
+				currentSa.ImagePullSecrets = append(currentSa.ImagePullSecrets, newPullSecret)
+				return nil
+			})
+			if err != nil {
+				return err
+			}
+		}
+	}
+	return nil
+}
```

Comment:
@StevenTobin @wei-lee Did we not decide to remove the SA linking?

Reply:
Yeah I removed where this function was called but not the function itself, just an oversight, I'll remove it now.